pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -0,0 +1,566 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	"github.com/tektoncd/pipeline/pkg/apis/version"
+
+	"github.com/tektoncd/pipeline/pkg/reconciler/pipeline/dag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/kmeta"
+)
+
+const (
+	// PipelineTasksAggregateStatus is a param representing aggregate status of all dag pipelineTasks
+	PipelineTasksAggregateStatus = "tasks.status"
+	// PipelineTasks is a value representing a task is a member of "tasks" section of the pipeline
+	PipelineTasks = "tasks"
+	// PipelineFinallyTasks is a value representing a task is a member of "finally" section of the pipeline
+	PipelineFinallyTasks = "finally"
+)
+
+// +genclient
+// +genclient:noStatus
+// +genreconciler:krshapedlogic=false
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Pipeline describes a list of Tasks to execute. It expresses how outputs
+// of tasks feed into inputs of subsequent tasks.
+// +k8s:openapi-gen=true
+type Pipeline struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state of the Pipeline from the client
+	// +optional
+	Spec PipelineSpec `json:"spec"`
+}
+
+var _ kmeta.OwnerRefable = (*Pipeline)(nil)
+
+// PipelineMetadata returns the Pipeline's ObjectMeta, implementing PipelineObject
+func (p *Pipeline) PipelineMetadata() metav1.ObjectMeta {
+	return p.ObjectMeta
+}
+
+// PipelineSpec returns the Pipeline's Spec, implementing PipelineObject
+func (p *Pipeline) PipelineSpec() PipelineSpec {
+	return p.Spec
+}
+
+// // Copy returns a deep copy of the Pipeline, implementing PipelineObject
+// func (p *Pipeline) Copy() PipelineObject {
+// 	return p.DeepCopy()
+// }
+
```

Comment:
should this be uncommented?

Reply:
I noticed in [`v1beta1/task_types.go`](https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/task_types.go/#L71-L74) there is the deepcopy while in v1 task_types there is not. Thought we should do this as we don't have the interface moved?

Reply:
I'm not sure what this is for actually but if it's not needed let's remove it rather than leaving it commented. we can always add it later.